cmd/snap/cmd_set_test.go

```diff
@@ -127,21 +127,51 @@ func (s *snapSetSuite) TestSnapSetFailParsingWithStrictJSON(c *check.C) {
 	c.Assert(err, check.ErrorMatches, "failed to parse JSON:.*")
 }
 
-func (s *snapSetSuite) TestSnapSetFailJsonKeysValidation(c *check.C) {
+func (s *snapSetSuite) TestSnapSetJSONStructureValidationUnhappy(c *check.C) {
+
+	invalidJson := []string{
+		`key={"key0":"val0",:"val1","key2":{"key3":"val2","key4":"val3","key5":{"key6":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}}}`,
+		`key={"key0":"val0","key1":"val1","key2"{"key3":"val2","key4":"val3",` +
+			`"key5":{"key6":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}},` +
+			`"key11":[{"key12":"val7"},{"key13":"val8"},{"key14":["val9","val10"]}]}`,
+		`key={"key0": ["a", "b" "c"]}`,
+		`key=[{"key1": "a"} {"key2": "b"}]`,
+		`key=[{"key1": "a"}, {"key2": "b}]`,
```

Comment:
I can return an error from the case in which I identify that the current type is `[]map[string]interface{}`, please let me know if this will fit this purpose.